relay-server/src/metrics_extraction/spans/mod.rs

```diff
@@ -303,6 +310,26 @@ fn sanitized_span_description(
     Some(sanitized)
 }
 
+/// Trims the given span description to ensure
+/// [`relay-metrics::AggregatorService`] doesn't drop it for being too long.
+///
+/// If the description is short, it remains unchanged. If it's long, this method
+/// truncates it to the maximum allowed length and sets the last character to
+/// `*`. The maximum length is determined by [`MaxChars::TagValue`].
+///
+/// Note [`relay-metrics::AggregatorService`] is configured independently and
+/// doesn't use this type; this method uses it for simplicity, and because at
+/// the time of writing this the values are the same. If the limit this type
+/// provides is longer, this method may be ineffective.
+fn truncate_span_description(description: String) -> String {
+    let mut truncated = description;
+    if bytecount::num_chars(truncated.as_bytes()) > MaxChars::TagValue.limit() {
+        truncated.truncate(MaxChars::TagValue.limit() - 1);
```

Comment:
`truncate` takes a number of bytes as input, but on line 326, we interpret `MaxChars::TagValue.limit()` as a number of characters. But what's worse is that `truncate` [might panic](https://doc.rust-lang.org/std/string/struct.String.html#panics-3), as @TBS1996 said.

You could use a function like this to truncate safely: https://github.com/getsentry/relay/blob/83a435e0754df148184c4ad642fb8a33eba3a469/relay-monitors/src/lib.rs#L145-L149